examples/composite_dialog/my_app_window/mod.rs

```diff
@@ -0,0 +1,64 @@
+// Since Gtk4 v4.10 gtk4::Dialog is deprecated and gtk4-rs's examples minimum version is v4.10
+#[allow(deprecated)]
+mod imp;
+
+use gtk::{glib, prelude::*, subclass::prelude::ObjectSubclassIsExt};
+
+glib::wrapper! {
+    pub struct MyAppWindow(ObjectSubclass<imp::MyAppWindow>)
+        @extends gtk::Widget, gtk::Window, gtk::ApplicationWindow;
+}
+
+#[gtk::template_callbacks]
+impl MyAppWindow {
+    pub fn new<P: IsA<gtk::Application>>(app: &P) -> Self {
+        glib::Object::builder().property("application", app).build()
+    }
+
+    /// Callback handler for notify::label signal.
+    ///
+    /// When counter property reach 3, a dialog pops up and present the user
+    /// with 2 choices: Set the counter to 6 or reset the counter to 0.
+    #[template_callback]
+    fn popup_dialog(&self, _p: &glib::ParamSpec) {
```

Comment:
I would rename the handler to `on_counter_notify`

Reply:
`popup_dialog` renamed `on_counter_notify`.